### PR TITLE
Fixed Apache RAT exclusions

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Get rid of unused Apache RAT exclusions as commented on incubator mailing list vote for previous release.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -615,7 +615,8 @@ flexible messaging model and an intuitive client API.</description>
             <exclude>**/*.crt</exclude>
             <exclude>**/*.key</exclude>
             <exclude>**/*.csr</exclude>
-            <exclude>**/proto/*.java</exclude>
+            <exclude>src/main/java/org/apache/bookkeeper/mledger/proto/MLDataFormats.java</exclude>
```
Comment (rdhabalia) [inline pom.xml]: formatting?

File: pom.xml [hunk 2]
```diff
@@ -686,48 +687,19 @@ flexible messaging model and an intuitive client API.</description>
...
-
-            <exclude>**/*.patch</exclude>
-            <exclude>.github/*.md</exclude>
-
             <exclude>src/assemble/LICENSE.bin.txt</exclude>
             <exclude>src/assemble/NOTICE.bin.txt</exclude>
 
             <!-- Protobuf generated files -->
-            <exclude>**/proto/*.java</exclude>
+            <exclude>src/main/java/org/apache/bookkeeper/mledger/proto/MLDataFormats.java</exclude>
```
Comment (rdhabalia) [inline pom.xml]: formatting?

File: pom.xml [hunk 3]
```diff
@@ -615,7 +615,8 @@ flexible messaging model and an intuitive client API.</description>
             <exclude>**/*.crt</exclude>
             <exclude>**/*.key</exclude>
             <exclude>**/*.csr</exclude>
-            <exclude>**/proto/*.java</exclude>
+            <exclude>src/main/java/org/apache/bookkeeper/mledger/proto/MLDataFormats.java</exclude>
+            <exclude>src/main/java/org/apache/pulsar/common/api/proto//PulsarApi.java</exclude>
```
Comment (rdhabalia) [inline pom.xml]: `//PulsarApi.java` additional `/`

File: pom.xml (hunk 1 above)
Comment (dave2wave) [reply pom.xml]: I checked. These are generated and a license is not generated. This is ok.

File: pom.xml (hunk 2 above)
Comment (dave2wave) [reply pom.xml]: Perhaps a comment can be added to explain these excludes?

Comment (merlimat) [reply pom.xml]: Sure, I'll expand the comment to be a bit more clear.

@rdhabalia What do you mean by formatting? The indentation looks ok to me.

Comment (rdhabalia) [reply pom.xml]: > The indentation looks ok to me.

Yes, I think in review-window, indentation didn't appear correctly. But it seems correct. So, please ignore it.

**Comment by merlimat** [comment]
Just to make it clear: I have removed all the exclusions for files that are not included in the src tarball. That includes all IDEs files, Makefile and so on. 

Probably RAT will not pass on a dirty GIT working directory, but it does pass when unpacking the tarball of the release.

**Pull request merged**